### ISSUE TITLE
[Bug]: SAM callback fails to process losses in multi-data and multi-task pipelines.

### ISSUE THREAD

**Issue by melo-gonzo** [IntelLabs/matsciml#184]
### Expected behavior

SAM would be able to process a single loss per optimizer.

### Actual behavior

The losses returned from `_compute_losses()` are a nested dictionary with dataset and task specific mapping, e.g.:
```python
>>> loss = task._compute_losses(self.batch)
>>> loss{'IS2REDataset': {'regression': {'loss': tensor(58.0430, grad_fn=<AddBackward0>), 'log': {'energy_init': tensor(32.1578, grad_fn=<MseLossBackward0>), 'energy_relaxed': tensor(25.8853, grad_fn=<MseLossBackward0>)}}}, 'S2EFDataset': {'regression': {'loss': tensor(265.4653, grad_fn=<AddBackward0>), 'log': {'force': tensor(0.1937, grad_fn=<MeanBackward0>), 'energy': tensor(265.2716, grad_fn=<MeanBackward0>)}}}}
```

### Steps to reproduce the problem

Add SAM callback to any of the examples in `./examples/tasks/multitask`

### Specifications

Latest

**Comment by melo-gonzo** [comment]
Closed in #185
